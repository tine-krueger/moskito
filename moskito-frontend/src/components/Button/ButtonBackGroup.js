import styled from 'styled-components/macro'
import Button from "./Button";

export default function ButtonBackGroup({text1, text2, onClick}) {
    return (
        <Buttons>
            <Button >{text1}</Button>
            <Button type="button" onClick={onClick}>{text2}</Button>
        </Buttons>
    )
}

const Buttons = styled.div`
    display: grid;
    justify-items: center;
    Button {
        margin: .5em;
        width: 10em;
    }
`