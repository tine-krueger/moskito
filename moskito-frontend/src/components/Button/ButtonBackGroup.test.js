import { render } from '@testing-library/react'
import ButtonBackGroup from './ButtonBackGroup'

const text1 = "Log in"
const text2 = "Zurück"
const onClick = jest.fn()

const component = <ButtonBackGroup
text1={text1}
text2={text2}
onClick={onClick}
/>

describe('Filter Campsite Checkbox', () => {
    it('renders correctly', () => {
        const { container } = render(component)
        expect(container).toMatchSnapshot()
    })

    it('shows Button texts', () => {
        const { getByText } = render(component)
        expect(getByText(/Log in/i)).toBeInTheDocument()
        expect(getByText(/Zurück/i)).toBeInTheDocument() 
    })

    it('does not show other texts', () => {
        const { queryByText } = render(component)
        expect(queryByText('Log out')).not.toBeInTheDocument()
        expect(queryByText(/Zuruck/i)).not.toBeInTheDocument() 
    })

})
