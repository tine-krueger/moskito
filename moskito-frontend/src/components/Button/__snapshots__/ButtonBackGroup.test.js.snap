// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Campsite Checkbox renders correctly 1`] = `
<div>
  <div
    class="ButtonBackGroup__Buttons-yrfass-0 gPvmVA"
  >
    <button
      class="Button__ButtonStyled-sc-1b627y-0 klBpgr"
    >
      Log in
    </button>
    <button
      class="Button__ButtonStyled-sc-1b627y-0 klBpgr"
    >
      Zur√ºck
    </button>
  </div>
</div>
`;
